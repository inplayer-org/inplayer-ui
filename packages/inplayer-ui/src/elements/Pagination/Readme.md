A component to render a pagination. It takes:
<br/> - `pageCount` - the total number of the pages
<br/> - `currentPage` - the current number of the page
<br/> - `selectPage` - action to select the page

```jsx static
import { Pagination } from '@inplayer-org/inplayer-ui';

const paginateData = {
  total: 50,
  data: [{ id: 1, name: 'foo' }, { id: 2, name: 'bar' }],
};

const totalPages = Math.ceil(paginateData.total / 5);

class PaginationExample extends React.Component {
  constructor() {
    super();

    this.state = {
      page: 0,
    };

    this.onPageChange = this.onPageChange.bind(this);
  }

  onPageChange(data) {
    this.setState({ page: data.selected });
  }

  render() {
    return (
      <Pagination
        pageCount={totalPages}
        currentPage={this.state.page}
        pageRangeDisplayed={5}
        marginPagesDisplayed={5}
        previousLabel="PREV"
        nextLabel="NEXT"
        breakLabel="..."
        onPageChange={this.onPageChange}
      />
    );
  }
}
<PaginationExample />;
```

### Demo

```jsx
const paginateData = {
  total: 50,
  data: [{ id: 1, name: 'foo' }, { id: 2, name: 'bar' }],
};

const totalPages = Math.ceil(paginateData.total / 5);

class PaginationExample extends React.Component {
  constructor() {
    super();

    this.state = {
      page: 0,
    };

    this.onPageChange = this.onPageChange.bind(this);
  }

  onPageChange(data) {
    this.setState({ page: data.selected });
  }

  render() {
    return (
      <Pagination
        pageCount={totalPages}
        currentPage={this.state.page}
        pageRangeDisplayed={5}
        marginPagesDisplayed={5}
        previousLabel="PREV"
        nextLabel="NEXT"
        breakLabel="..."
        onPageChange={this.onPageChange}
      />
    );
  }
}
<PaginationExample />;
```
