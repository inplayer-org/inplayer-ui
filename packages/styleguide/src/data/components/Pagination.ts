import { ComponentData } from "../types";

const usage = `
  import { Pagination } from '@inplayer-org/inplayer-ui';

  const paginateData = {
    total: 50,
    data: [{ id: 1, name: 'foo' }, { id: 2, name: 'bar' }],
  };

  const totalPages = Math.ceil(paginateData.total / 5);

  class PaginationExample extends React.Component {
    constructor() {
      super();

      this.state = {
        page: 0,
      };

      this.onPageChange = this.onPageChange.bind(this);
    }

    onPageChange(data) {
      this.setState({ page: data.selected });
    }

    render() {
      return (
        <Pagination
          pageCount={totalPages}
          currentPage={this.state.page}
          pageRangeDisplayed={5}
          marginPagesDisplayed={5}
          previousLabel="PREV"
          nextLabel="NEXT"
          breakLabel="..."
          onPageChange={this.onPageChange}
        />
      );
    }
  }
  <PaginationExample />;
`

const demoCode = `
  class PaginationExample extends React.Component {
    constructor() {
      super();

      this.state = {
        page: 0,
        paginateData: {
          total: 50,
          data: [{ id: 1, name: 'foo' }, { id: 2, name: 'bar' }],
        },
      };

      this.onPageChange = this.onPageChange.bind(this);
    }

    onPageChange(data) {
      this.setState({ page: data.selected });
    }

    render() {
      return (
        <Pagination
          pageCount={Math.ceil(this.state.paginateData.total / 5)}
          currentPage={this.state.page}
          pageRangeDisplayed={5}
          marginPagesDisplayed={5}
          previousLabel="PREV"
          nextLabel="NEXT"
          breakLabel="..."
          onPageChange={this.onPageChange}
        />
      );
    }
  }
`;

const Pagination: ComponentData = {
  description: `A component to render a pagination. It takes: 
  - pageCount - the total number of the pages 
  - currentPage - the current number of the page 
  - selectPage - action to select the page
  `,
  path: `src/elements/Pagination/index.js`,
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'breakLabel', type: 'string', default: 'Required' },
    { name: 'currentPage', type: 'number', default: 'Required' },
    { name: 'marginPagesDisplayed', type: 'number', default: 'Required' },
    { name: 'nextLabel', type: 'string', default: 'Required' },
    { name: 'onPageChange', type: 'function', default: 'Required' },
    { name: 'pageCount', type: 'number', default: 'Required' },
    { name: 'pageRangeDisplayed', type: 'number', default: 'Required' },
    { name: 'previousLabel', type: 'string', default: 'Required' },
    { name: 'activeClassName', type: 'string', default: 'selected' },
    { name: 'activeLinkClassName', type: 'string', default: 'selected' },
    { name: 'breakClassName', type: 'string', default: 'break' },
    { name: 'className', type: 'string'},
    { name: 'containerClassName', type: 'string'},
    { name: 'disableInitialCallback', type: 'boolean', default: 'false'},
    { name: 'disabledClassName', type: 'string', default: 'disabled'},
    { name: 'extraAriaContext', type: 'string'},
    { name: 'hrefBuilder', type: 'function', default: 'Function'},
    { name: 'initialPage', type: 'number', default: '0'},
    { name: 'nextClassName', type: 'string', default: 'next'},
    { name: 'nextLinkClassName', type: 'string', default: 'next'},
    { name: 'pageClassName', type: 'string'},
    { name: 'pageLinkClassName', type: 'string'},
    { name: 'previousClassName', type: 'string', default: 'previous'},
    { name: 'previousLinkClassName', type: 'string', default: 'previous'},
    { name: 'style', type:'Object', default: '{}' },
  ]
}

export default Pagination;