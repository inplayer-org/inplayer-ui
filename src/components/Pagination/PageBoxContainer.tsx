import React, { ButtonHTMLAttributes } from 'react';
import {
  AnalyticsComponent,
  AnalyticsEvents,
  AnalyticsComponentType,
  AnalyticsProps,
} from '../../analytics';
import { PageBox, BoxStyles } from './styled';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement>, AnalyticsProps, BoxStyles {}
const PageBoxContainer = ({ onClick, tag = '', ...rest }: Props) => (
  <AnalyticsComponent>
    {({ pages, tracker, merchantId, ip }) => (
      <PageBox
        onClick={(e) => {
          onClick?.(e);
          if (tag) {
            tracker.track({
              event: AnalyticsEvents.CLICK,
              type: AnalyticsComponentType.PAGINATION,
              tag,
              pages,
              merchantId,
              ip,
            });
          }
        }}
        {...rest}
      />
    )}
  </AnalyticsComponent>
);

export default PageBoxContainer;
