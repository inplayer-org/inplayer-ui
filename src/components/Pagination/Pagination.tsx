import React, { useState, useEffect } from 'react';
import { FaAngleDoubleLeft, FaAngleDoubleRight, FaAngleLeft, FaAngleRight } from 'react-icons/fa';
import { PageBox, PaginationContainer } from './styled';

type PaginationProps = {
  /**
   * Page change handler. Receive pageNumber as arg
   */
  onPageChange: (pageNumber: number) => void;
  /**
   * Total count of items which you are going to display
   */
  totalItems: number;
  /**
   * Start page / Active page
   */
  startPage?: number;
  /**
   * Range of pages in paginator, exclude navigation blocks (prev, next, first, last pages)
   */
  numberOfPagesDisplayed?: number;
  /**
   * Count of items per page
   */
  itemsPerPage?: number;
};

// Kibana has limit of 9990 pages
const pagesLimit = 9990;

const Pagination = ({
  onPageChange,
  numberOfPagesDisplayed = 5,
  totalItems,
  startPage = 1,
  itemsPerPage = 15,
}: PaginationProps) => {
  const [activePage, setActivePage] = useState(startPage);
  const [visiblePages, setVisiblePages] = useState<number[]>([]);

  let totalPages = Math.ceil(totalItems / itemsPerPage);
  totalPages = Math.min(totalPages, pagesLimit);
  const numberOfPagesDisplayedScaled =
    totalPages < numberOfPagesDisplayed ? totalPages : numberOfPagesDisplayed;
  const isVisibleLess = !visiblePages.some((index) => index === 1);
  const isVisibleMore = !visiblePages.some((index) => index === totalPages);

  useEffect(() => {
    let minVisiblePage = 0;
    // this won't work if user wants to show less than 3 pages
    if (startPage + 1 >= totalPages) {
      minVisiblePage = totalPages - numberOfPagesDisplayedScaled + 1;
    } else if (startPage - 1 <= 1) {
      minVisiblePage = 1;
    } else minVisiblePage = startPage - 1;

    // Create the pages that are visible for selection
    const initialPagesVisible = Array.from(Array(numberOfPagesDisplayedScaled).keys()).map(
      (i) => i + minVisiblePage
    );

    setVisiblePages(initialPagesVisible);
  }, [totalItems, itemsPerPage, numberOfPagesDisplayed]);

  useEffect(() => {
    setActivePage(startPage);
  }, [startPage]);

  const onPageClick = (pageNumber: number) => () => {
    const indexOfPage = visiblePages.indexOf(pageNumber);
    let offset = 0;
    if (indexOfPage === 0 && isVisibleLess) {
      offset = -1;
    } else if (indexOfPage === numberOfPagesDisplayedScaled - 1 && isVisibleMore) {
      offset = 1;
    }

    if (offset !== 0) {
      setVisiblePages((pages) => pages.map((page) => page + offset));
    }
    setActivePage(pageNumber);
    onPageChange(pageNumber);
  };

  const onDotsClick = (more: boolean) => () => {
    let offset = 0;
    const deduction = numberOfPagesDisplayedScaled === 3 ? 2 : 3;
    if (!more) {
      if (visiblePages[0] - deduction >= 1) {
        offset = 0 - deduction;
      } else offset = 3 - visiblePages[0] - 2;
      setActivePage(visiblePages[0] - 1);
      onPageChange(visiblePages[0] - 1);
    } else {
      if (visiblePages[numberOfPagesDisplayedScaled - 1] + deduction <= totalPages) {
        offset = deduction;
      } else {
        offset = totalPages - visiblePages[numberOfPagesDisplayedScaled - 1];
      }
      setActivePage(visiblePages[numberOfPagesDisplayedScaled - 1] + 1);
      onPageChange(visiblePages[numberOfPagesDisplayedScaled - 1] + 1);
    }

    setVisiblePages((pages) => pages.map((page) => page + offset));
  };

  const goToEnd = () => {
    setActivePage(totalPages);
    const endPages = Array.from(
      Array(Math.min(Math.ceil(totalItems / itemsPerPage), numberOfPagesDisplayedScaled)).keys()
    )
      .map((number) => totalPages - number)
      .reverse();
    setVisiblePages(endPages);
    onPageChange(totalPages);
  };

  const goToStart = () => {
    setActivePage(1);
    const startPages = Array.from(
      Array(Math.min(Math.ceil(totalItems / itemsPerPage), numberOfPagesDisplayedScaled)).keys()
    ).map((number) => number + 1);
    setVisiblePages(startPages);
    onPageChange(1);
  };

  if (totalItems === 0) {
    return null;
  }

  return (
    <PaginationContainer>
      <PageBox type="button" disabled={activePage === 1} onClick={goToStart}>
        <FaAngleDoubleLeft />
      </PageBox>
      <PageBox type="button" disabled={activePage === 1} onClick={onPageClick(activePage - 1)}>
        <FaAngleLeft />
      </PageBox>
      {!visiblePages.some((index) => index === 1) && (
        <PageBox type="button" hideBorder onClick={onDotsClick(false)}>
          •••
        </PageBox>
      )}
      {visiblePages.map((index) => (
        <PageBox
          type="button"
          selected={activePage === index}
          key={index}
          onClick={onPageClick(index)}
        >
          {index}
        </PageBox>
      ))}
      {!visiblePages.some((index) => index === totalPages) && (
        <PageBox type="button" hideBorder onClick={onDotsClick(true)}>
          •••
        </PageBox>
      )}
      <PageBox
        type="button"
        disabled={activePage === totalPages}
        onClick={onPageClick(activePage + 1)}
      >
        <FaAngleRight />
      </PageBox>
      <PageBox type="button" disabled={activePage === totalPages} onClick={goToEnd}>
        <FaAngleDoubleRight />
      </PageBox>
    </PaginationContainer>
  );
};

Pagination.defaultProps = {
  startPage: 1,
  numberOfPagesDisplayed: 5,
  itemsPerPage: 15,
};

export default Pagination;
