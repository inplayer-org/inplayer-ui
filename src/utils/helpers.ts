import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
