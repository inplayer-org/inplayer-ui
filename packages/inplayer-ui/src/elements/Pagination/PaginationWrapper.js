// @flow
import styled from 'styled-components';

import colors from 'config/colors';
import { uiColors, fontSizes, fontWeights } from 'utils';

const PaginationWrapper = styled.div`
  border: 1px solid ${colors.gray};
  border-radius: 3px;
  background: ${colors.white};
  font-size: ${fontSizes('small')};
  line-height: 30px;
  overflow: hidden;
  box-sizing: border-box;
  vertical-align: top;
  cursor: pointer;
  display: inline-flex;
  margin: 1rem auto;

  ul {
    padding-inline-start: 0;
    margin: 0;
    font-weight: ${fontWeights('light')};
  }

  li {
    text-decoration: none;
    display: inline-flex;

    a {
      color: ${uiColors('text.light')};
      padding: 0.0625rem 1rem 0.0625rem;

      &:focus {
        outline: none;
      }
    }

    &.selected {
      background: ${uiColors('primary.main')};
      pointer-events: none;

      a {
        font-weight: ${fontWeights('semibold')};
        color: ${colors.white};
      }
    }

    &.disabled {
      color: ${uiColors('text.disabled')};

      a {
        color: ${uiColors('text.disabled')} !important;
      }
    }

    &.previous {
      border-right: 1px solid ${colors.gray};

      a {
        color: ${uiColors('text.light')};
        padding: 0.0625rem 1rem 0.0625rem;

        &::before {
          content: '⟵';
          padding-right: 0.5rem;
          position: relative;
          top: -1px;
          left: 0;
          transition: ease all 200ms;
        }
      }
    }

    &.next {
      border-left: 1px solid ${colors.gray};

      a {
        color: ${uiColors('text.light')};
        padding: 0.0625rem 1rem 0.0625rem;

        &::after {
          content: '⟶';
          padding-left: 0.5rem;
          position: relative;
          top: -1px;
          right: 0;
          transition: ease all 200ms;
        }
      }
    }

    &.previous:hover:not(.disabled) {
      background: ${colors.lightGray};

      a {
        color: ${uiColors('text.main')};

        &::before {
          left: -3px;
        }
      }
    }

    &.next:hover:not(.disabled) {
      background: ${colors.lightGray};

      a {
        color: ${uiColors('text.main')};

        &::after {
          right: -3px;
        }
      }
    }

    &:hover:not(.disabled) {
      color: ${uiColors('text.main')};
      background: ${colors.lightGray};
    }
  }
`;

export default PaginationWrapper;
