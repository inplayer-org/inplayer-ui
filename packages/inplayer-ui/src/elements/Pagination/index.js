// @flow
import React from 'react';
import ReactPaginate from 'react-paginate';
import styled from 'styled-components';

import PaginationWrapper from './PaginationWrapper';

type Props = {
  pageCount: number,
  currentPage: number,
  pageRangeDisplayed: number,
  marginPagesDisplayed: number,
  previousLabel: string,
  nextLabel: string,
  breakLabel: string,
  onPageChange: (data: { selected: number }) => any,
  breakClassName?: string,
  initialPage?: number,
  disableInitialCallback?: boolean,
  containerClassName?: string,
  pageClassName?: string,
  pageLinkClassName?: string,
  activeClassName?: string,
  activeLinkClassName?: string,
  previousClassName?: string,
  nextClassName?: string,
  previousLinkClassName?: string,
  nextLinkClassName?: string,
  disabledClassName?: string,
  hrefBuilder?: () => string,
  extraAriaContext?: string,
  style?: Object,
  className?: string,
};

const PaginationPositioning = styled.div`
  display: block;
  margin: 0 auto;
  text-align: center;
`;

const Pagination = ({ style, className, currentPage, ...rest }: Props) => (
  <PaginationPositioning style={style} className={className}>
    <PaginationWrapper>
      <ReactPaginate forcePage={currentPage} {...rest} />
    </PaginationWrapper>
  </PaginationPositioning>
);

Pagination.defaultProps = {
  breakClassName: 'break',
  initialPage: 0,
  disableInitialCallback: false,
  containerClassName: '',
  pageClassName: '',
  pageLinkClassName: '',
  activeClassName: 'selected',
  activeLinkClassName: 'selected',
  previousClassName: 'previous',
  nextClassName: 'next',
  previousLinkClassName: 'previous',
  nextLinkClassName: 'next',
  disabledClassName: 'disabled',
  hrefBuilder: () => '',
  extraAriaContext: '',
  style: {},
  className: '',
};

export default Pagination;
