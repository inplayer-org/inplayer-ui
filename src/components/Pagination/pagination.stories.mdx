import { useState } from 'react';
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Pagination from './'

<Meta title="Pagination" component={<Pagination/>} />

# Pagination

Very easy to use. Just provide `props` with total amount of things that you want to display on the page.

```jsx
import { Pagination } from '@inplayer-org/inplayer-ui';

const Page = () => (
  <Pagination />
)
```

### Props 

<Props of={Pagination}/>

### Pagination example
<Preview withSource="open">
  <Story name="pagination">
    {() => {
        const [page, setPage] = useState(1);
        const onPageChange = (data) => setPage(data.selected);
        return(
            <Pagination
                onPageChange={(data) => onPageChange(data)}
                totalItems={50}
                itemsPerPage={5}
                numberOfPagesDisplayed={5}
                startPage={page}
            />
        )
    }}
  </Story>
</Preview>

