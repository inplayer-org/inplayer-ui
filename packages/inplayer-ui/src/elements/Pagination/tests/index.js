import { generateSnapshots } from 'packages/inplayer-ui/src/elements/Pagination/tests/helpers';

import Pagination from '../index';

generateSnapshots(Pagination);
