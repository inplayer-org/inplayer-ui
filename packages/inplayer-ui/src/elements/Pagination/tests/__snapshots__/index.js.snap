// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
.c1 {
  border: 1px solid #d7dde5;
  border-radius: 3px;
  background: #fff;
  font-size: 0.875rem;
  line-height: 30px;
  overflow: hidden;
  box-sizing: border-box;
  vertical-align: top;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 1rem auto;
}

.c1 ul {
  padding-inline-start: 0;
  margin: 0;
  font-weight: 300;
}

.c1 li {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 li a {
  color: #7d7e7f;
  padding: 0.0625rem 1rem 0.0625rem;
}

.c1 li a:focus {
  outline: none;
}

.c1 li.selected {
  background: #07AAE6;
  pointer-events: none;
}

.c1 li.selected a {
  color: #fff;
}

.c1 li.disabled {
  color: #c8ccd4;
}

.c1 li.disabled a {
  color: #c8ccd4 !important;
}

.c1 li.previous {
  border-right: 1px solid #d7dde5;
}

.c1 li.previous a {
  color: #7d7e7f;
  padding: 0.0625rem 1rem 0.0625rem;
}

.c1 li.previous a::before {
  content: '⟵';
  padding-right: 0.5rem;
  position: relative;
  top: -1px;
  left: 0;
  -webkit-transition: ease all 200ms;
  transition: ease all 200ms;
}

.c1 li.next {
  border-left: 1px solid #d7dde5;
}

.c1 li.next a {
  color: #7d7e7f;
  padding: 0.0625rem 1rem 0.0625rem;
}

.c1 li.next a::after {
  content: '⟶';
  padding-left: 0.5rem;
  position: relative;
  top: -1px;
  right: 0;
  -webkit-transition: ease all 200ms;
  transition: ease all 200ms;
}

.c1 li.previous:hover:not(.disabled) {
  background: #F6F8FB;
}

.c1 li.previous:hover:not(.disabled) a {
  color: #3e4348;
}

.c1 li.previous:hover:not(.disabled) a::before {
  left: -3px;
}

.c1 li.next:hover:not(.disabled) {
  background: #F6F8FB;
}

.c1 li.next:hover:not(.disabled) a {
  color: #3e4348;
}

.c1 li.next:hover:not(.disabled) a::after {
  right: -3px;
}

.c1 li:hover:not(.disabled) {
  color: #3e4348;
  background: #F6F8FB;
}

.c0 {
  display: block;
  margin: 0 auto;
  text-align: center;
}

<div
  className="c0"
  style={Object {}}
>
  <div
    className="c1"
  >
    <ul
      className=""
    >
      <li
        className="previous disabled"
      >
        <a
          aria-disabled="true"
          className="previous"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Previous
        </a>
      </li>
      <li
        className=" selected"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          className=" selected"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          1
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          aria-label="Page 2"
          className=""
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          2
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          aria-label="Page 3"
          className=""
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          3
        </a>
      </li>
      <li
        className="break"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          ...
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          aria-label="Page 8"
          className=""
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          8
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          aria-label="Page 9"
          className=""
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          9
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          aria-label="Page 10"
          className=""
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          10
        </a>
      </li>
      <li
        className="next"
      >
        <a
          aria-disabled="false"
          className="next"
          href=""
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;
